File: src/Dtos/CommentDto.cs
namespace Posts_Website.Dtos
{
	public record class CommentDto
	(
		Guid Id,
		Guid PostId,
		string Content,
		string WrittenAt,
		UserDto User
	);	
	
	public record class CommentDetailsDto
	(
		Guid Id,
		Guid PostId,
		string Content,
		string WrittenAt,
		UserDto User,
		PostDto Post
	);

	public record class CommentFormDto
	(
		string Content
	);
}

File: Client/src/typescript/dtos/CommentDto.ts
import { PostDto } from "./PostDto";
import { UserDto } from "./UserDto";

export interface CommentDto {
    id: string,
    postId: string,
    content: string,
    writtenAt: string,
    user: UserDto
}

export interface CommentDetailsDto extends CommentDto {
    post: PostDto
}

export interface CommentFormDto {
    content: string
}